source/lac/petsc_vector_base.cc

```diff
@@ -198,7 +198,7 @@ namespace PETScWrappers
       OutType
       operator*() const
       {
-        return static_cast<OutType>(*m_iterator);
+        return static_cast<OutType>(std::real(*m_iterator));
```

Comment:
This just puts the whole idea of the `ConvertingIterator` to lies. At least it deserves a TODO :-)